rust/signer/src/lib.rs

```diff
@@ -341,4 +342,31 @@ mod tests {
 
         assert!(is_valid);
     }
+
+    #[test]
+    fn test_substrate_sign_with_ref() {
+        let msg: String = b"Build The Future".to_hex();
+        let data_pointer = decrypt_data_ref(ENCRYPTED_SEED, String::from(PIN)).unwrap();
+        let signature_by_ref = substrate_brainwallet_sign_with_ref(data_pointer, &msg).unwrap();
+        let is_valid = schnorrkel_verify(SURI, &msg, &signature_by_ref).unwrap();
+        assert!(is_valid);
+    }
+
+    #[test]
+    #[should_panic]
+    fn test_substrate_sign_with_ref_destroyed() {
+        let msg: String = b"Build The Future".to_hex();
+        let data_pointer = decrypt_data_ref(ENCRYPTED_SEED, String::from(PIN)).unwrap();
+        destroy_data_ref(data_pointer);
+        let signature_by_ref = substrate_brainwallet_sign_with_ref(data_pointer, &msg).unwrap();
+        let is_valid = schnorrkel_verify(SURI, &msg, &signature_by_ref).unwrap();
+        assert!(!is_valid);
```

Comment:
And tried by multiple times, it does not always get `malloc` error.

Reply:
The behavior will be undefined, since it is accessing dropped memory, but it is not a guaranteed fault.